### ISSUE TITLE
Developer's guide?

### ISSUE THREAD

**Issue by cwsmith** [mfem/mfem#223]
@tzanio @v-dobrev,
We (@KazemKamran and I) would like to contribute the PUMI integration code to this repo but did not find a developer's guide on the recommended process.  Is a guide available?

It appears that forks are created, then (eventually) pull requests are issued that trigger code review and automated testing.  Is that the suggested process?

**Comment by tzanio** [comment]
@cwsmith, @KazemKamran: Apologies for not having clear guidelines, we are working on a `CONTRIBUTING.md`... Here is a short overview:

1. We prefer if you create the new feature branch inside the MFEM organization as opposed to in a fork (so the whole team can collaborate on it). Both of you are members of the organization, so you should be able to do that.
1. The typical feature branch name is `new-feature-dev`, e.g. `pumi-dev`.
1. When you want other people to start looking / commenting on the code, create a pull request towards `mfem:master`.
1. Pull request names are typically `Description [new-feature-dev]`, e.g. `Parallel Unstructured Mesh Infrastructure (PUMI) integration [pumi-dev]`.
1. Add a description, appropriate labels and assign yourself to the PR. We will add reviewers as appropriate.
1. List outstanding TODO items in the description, see e.g. #222.
1. Track the Travis/Appveyor builds for errors.
1. Once the review is complete and the PR is approved, it is manually merged in the `next` branch and the `in-next` label is added to the PR.
1. After a week of testing in `next`, both on GitHub, as well as internally at LLNL, the original PR is merged into `master` (provided there are no issues).
1. The `master` branch should always be of release quality.

Does this make sense? I can elaborate if something is not clear.

Thanks for your contribution!

Tzanio

**Comment by cwsmith** [comment]
OK.  This sounds good. Thank you.

Is there a rule of thumb for PR granularity?  For example, would support for loading a partitioned PUMI mesh then creating a MFEM mesh from it constitute a feature worthy of a merge?  Given item 6 (TODO list), I assume that this would be fine.

**Comment by tzanio** [comment]
Yes, that sounds like a good PR. Smaller granularity is probably better.

**Comment by tzanio** [comment]
Closing this issue, as we will soon have a PR with a more descriptive developer guide.

**Comment by tzanio** [comment]
See #410
